prereise/gather/demanddata/transportation_electrification/mileage.py

```diff
@@ -0,0 +1,201 @@
+# up to line 117 done
+import numpy as np
+import pandas as pd
+from scipy.io import loadmat
+
+from prereise.gather.demanddata.transportation_electrification import const
+
+kate = 1
+battery_size_list = []
+
+# can use PHEV and EV
+
+
+def get_input_day(month_days: list[int], month_first_day: list[int]) -> np.ndarray:
+    """Determine month of each day and day of week for any given day in the year.
+
+    :param list month_days: a list of integers where each value in the list represents
+        the number of days in that month (0 indexed).
+    :param list month_first_day: a list of integers where each value in the list
+        represents the day of the week of the first day of that month (0 indexed).
+    :return: (*np.ndarray*) -- first list stores the day of the week for each day of
+        the year, second list stores the month the day is in for each day of the year.
+
+    """
+    # initalize day counter
+    day_count = 0
+
+    # initialize
+    input_day = np.zeros(365, int)
+
+    # iterate over 12 months
+    for i in range(len(month_days)):
+        # initialize first day of each month for use in the next for loop
+        k = month_first_day[i]
+
+        # Iterate over the day of each month
+        for j in range(month_days[i]):
+
+            # if k < 6: weekday, if k >= 6: weekend
+            if k < 6:
+                input_day[day_count] = 2
+            elif k >= 6:
+                input_day[day_count] = 1
+
+            # restarts the week counter independent of the intial day
+            if k == 7:
+                k = 1
+            else:
+                k += 1
+
+            # increment day counter
+            day_count += 1
+
+    return input_day

```

Comment:
This function does not appear to be used. Reading the docstring, it seems that we could use the built-in attributes/methods of a pandas DatetimeIndex for this: e.g. the `dayofweek` and `month` attributes of the index will each return a series with the desired values for each data point.